tools/delete_candidate.php

```diff
@@ -65,10 +85,59 @@
 $query = "SELECT ID, Test_name, CommentID FROM flag WHERE SessionID in (" . 
          implode(" , ", $sessions) . ")";
 $instruments = $DB->pselect($query, array()); 
-
+//delete from genomic_candidate_files_rel
+$DB->delete("genomic_candidate_files_rel", array("CandID" => $DCCID));
+echo "----------------------delete genomic_candidate_files_rel--------";
+//delete from genomic_sample_candidate_rel
+$DB->delete("genomic_sample_candidate_rel", array("CandID" => $DCCID));
+echo "----------------------delete genomic_sample_candidate_rel--------";
+//delete from mri_scanner
+$DB->delete("mri_scanner", array("CandID" => $DCCID));
+echo "----------------------delete mri_scanner-------------------------";
+//delete from parameter_candidate
+$DB->delete("parameter_candidate", array("CandID" => $DCCID));
+echo "----------------------delete parameter candidate-----------------";
+//find session id
+$sessionIDs = $DB->pselect("SELECT ID 
+                FROM session 
+                WHERE candID=:candID",
+                array('candID' => $DCCID)
+                );
+
+
+//delete from media
+foreach ($sessionIDs as $sessionID) {
+$DB->delete("media", array("session_id" => $sessionID['ID']));
+};
+echo "----------------------delete media------------------------------";
+
+//delete from feedback_mri_comments
+foreach ($sessionIDs as $sessionID) {
+$fileIDs = $DB->pselect("SELECT FileID 
+                FROM files 
+                WHERE SessionID=:SessionID",
+                array('SessionID' => $sessionID['ID'])
+                );
+         foreach ($fileIDs as $fileID) {
+//delete from parameter_file
+         $DB->delete("parameter_file",
+                      array("FileID" => $fileID['FileID']));
+         
+//delete from feedback_mri_comments
+         $DB->delete("feedback_mri_comments", 
+                      array("FileID" => $fileID['FileID']));
+          };
+};
+echo "----------------------delete parameter_file----------------------";
+echo "----------------------delete feedback_mri_comments---------------";
+//delete from files
+foreach ($sessionIDs as $sessionID) {
+$DB->delete("files", array("SessionID" => $sessionID['ID']));
+};
+echo "----------------------delete files--------------------------------";
 //delete the sessions
 $DB->delete("session", array("CandID" => $DCCID));
```

Comment:
1. feedback_mri_comments might have a record for that session with a null value in FileID
2. mri_acquisition_dates has a foreign key on session
3. parameter_session also have  foreign key on session